src/alloc.c

```diff
@@ -803,7 +803,11 @@ static bool mi_try_new_handler(bool nothrow) {
   if (h==NULL) {
     _mi_error_message(ENOMEM, "out of memory in 'new'");      
     if (!nothrow) {
+#ifdef __EXCEPTIONS
```

Comment:
I think __EXCEPTIONS is GCC specific (maybe clang has it too). Perhaps we should turn the logic around: 
```C
#if defined(__GCC__) && !defined(__EXCEPTIONS)
abort();
#else 
throw std::bad_alloc();
#endif
```

Reply:
clang has it too so in your snippet `__GCC__` can be replaced by `__GNUC__`